src/renderers/shared/reconciler/__tests__/ReactCompositeComponent-test.js

```diff
@@ -1241,4 +1241,51 @@ describe('ReactCompositeComponent', function() {
 
   });
 
+  it('should warn when mutated props are passed', function() {
+
+    var container = document.createElement('div');
+
+    function Foo (props) {
+      React.Component.call(this, props);
+    }
+
+    Foo.prototype = Object.create(React.Component.prototype, {
+      constructor: {
+        value: Foo,
+      },
+    });
+
+    Foo.prototype.render = function() {
+      return <span />;
+    };
+
+    function Bar (props) {
```

Comment:
Do you need Foo at all? Why can't Bar extend React.Component directly?

Also, can you use ES6 class syntax here? It's simpler.